batcher/aligned-batcher/src/lib.rs

```diff
@@ -1189,39 +1199,67 @@ impl Batcher {
         proof_submitters: Vec<Address>,
         fee_params: CreateNewTaskFeeParams,
     ) -> Result<TransactionReceipt, BatcherError> {
-        info!("Creating task for: 0x{}", hex::encode(batch_merkle_root));
-
-        match try_create_new_task(
-            batch_merkle_root,
-            batch_data_pointer.clone(),
-            proof_submitters.clone(),
-            fee_params.clone(),
-            &self.payment_service,
-        )
-        .await
-        {
-            Ok(receipt) => Ok(receipt),
-            Err(BatcherSendError::TransactionReverted(err)) => {
-                // Since transaction was reverted, we don't want to retry with fallback.
-                warn!("Transaction reverted {:?}", err);
-
-                Err(BatcherError::TransactionSendError)
-            }
-            Err(_) => {
-                let receipt = try_create_new_task(
+        let result = retry_function(
+            || {
+                create_new_task_retryable(
                     batch_merkle_root,
-                    batch_data_pointer,
-                    proof_submitters,
-                    fee_params,
+                    batch_data_pointer.clone(),
+                    proof_submitters.clone(),
+                    fee_params.clone(),
+                    &self.payment_service,
                     &self.payment_service_fallback,
                 )
-                .await?;
-
-                Ok(receipt)
+            },
+            DEFAULT_MIN_RETRY_DELAY,
+            DEFAULT_BACKOFF_FACTOR,
+            DEFAULT_MAX_RETRIES,
+        )
+        .await;
+        match result {
+            Ok(receipt) => Ok(receipt),
+            Err(RetryError::Permanent(BatcherError::ReceiptNotFoundError)) => {
+                self.override_created_task(fee_params.gas_price).await;
+                Err(BatcherError::ReceiptNotFoundError)
             }
+            Err(_) => Err(BatcherError::TransactionSendError),
         }
     }
 
+    pub async fn override_created_task(&self, task_gas_price: U256) {
+        let from_address = self.batcher_signer.address();
+        let current_nonce = self
+            .batcher_signer
+            .get_transaction_count(from_address, None)
+            .await
+            .unwrap();
```

Comment:
let's not unwrap 👀

Reply:
maybe we should think about what to do if the cancelling of the transaction fails

Reply:
if there is an error here, we may want to try with fallback too

Reply:
Totally missed it, thank you! Trying it with the fallback sounds good too.

Reply:
also, we may want to make this function a retryable and do some retries.

Reply:
Yes, I will create a retryable function for this!